compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -2164,7 +2164,12 @@ public void visit(BLangFunction funcNode) {
                                                                    getOrigin(funcNode.name.value));
         funcSymbol.source = funcNode.pos.lineRange().filePath();
         funcSymbol.markdownDocumentation = getMarkdownDocAttachment(funcNode.markdownDocumentationAttachment);
-        SymbolEnv invokableEnv = SymbolEnv.createFunctionEnv(funcNode, funcSymbol.scope, env);
+        SymbolEnv invokableEnv;
+        if (env.node.getKind() == NodeKind.CLASS_DEFN || env.node.getKind() == NodeKind.OBJECT_TYPE) {
+            invokableEnv = SymbolEnv.createFunctionEnv(funcNode, funcSymbol.scope, env.enclEnv);
+        } else {
+            invokableEnv = SymbolEnv.createFunctionEnv(funcNode, funcSymbol.scope, env);
+        }
```

Comment:
Consider the following example
```ballerina
public class Class2 {
    int i = 0;

    public function foo(int i = i) { // error
        
    }
}
```
Here, the `enclEnv` of  `invokableEnv` created for the function `foo` is `pkgEnv`. IMO it should be `objectEnv` without fields.
@MaryamZi WDYT?

Reply:
Yeah, using the `pkgEnv` isn't correct.

I think what we discussed was that what we've use as the method's env would be an `env` that sees `self` instead of the fields like now? @KavinduZoysa, did we look into that?

Reply:
Changed, please check